Docs - rework group analytics doc (#9946)
PostHog/posthog.com: contents/docs/_snippets/setting-group-properties.mdx

Continuing with the previous example of using `company` as our group type, we'll add `company_name`, `date_joined`, and `subscription` as additional properties.

> > **Note:** You must include at least one group property for a group to be visible in the [Persons & Groups tab](https://app.posthog.com/persons).

Reviewer: ```suggestion
> **Note:** You must include at least one group property for a group to be visible in the [People and groups tab](https://app.posthog.com/persons).
```

<MultiLanguage>

```js-web

> // Option 1 (recommended):
> // Set properties in posthog.group()
> // Note that this has the side-effect that for the rest of the session
> // all subsequent events captured will be associated to the group

Reviewer: ```suggestion
// Option 1 (recommended): Set properties in posthog.group()
// This has the side-effect that all subsequent events in the session are associated to the group
```

]);
```

```ios_swift

> // Option 1 (recommended):
> // Set properties in group()
> // Note that this has the side-effect that for the rest of the session
> // all subsequent events captured will be associated to the group

Reviewer: ```suggestion
// Option 1 (recommended): Set properties in group()
// This has the side-effect that all subsequent events in the session are associated to the group
```

PostHogSDK.shared.group(
    type: "company", 
    key: "company_id_in_your_db",
    properties: [
        "name": "PostHog",
        "subscription": "subscription",
        "date_joined": "2020-01-23"
    ]
)

> // Option 2:
> // Set properties using capture
> // This method doesn't have the side-effect of associating all future events to the group

Reviewer: ```suggestion
// Option 2: Set properties using capture
// This method doesn't have the side-effect of associating the session's events to the group
```

PostHogSDK.shared.capture(
    event: "$groupidentify",
    properties: [
        "$group_type": "company",
        "$group_key": "company_id_in_your_db",
        "$group_set": [
            "name": "PostHog",
            "subscription": "subscription",
            "date_joined": "2020-01-23"
        ]
    ]
)
```

```android_kotlin

> // Option 1 (recommended):
> // Set properties in group()
> // Note that this has the side-effect that for the rest of the session
> // all subsequent events captured will be associated to the group

Reviewer: ```suggestion
// Option 1 (recommended): Set properties in group()
// This has the side-effect that all subsequent events in the session are associated to the group
```

PostHog.group(
    type = "company", 
    key = "company_id_in_your_db",
    properties = mapOf(
        "name" to "PostHog",
        "subscription" to "subscription",
        "date_joined" to "2020-01-23"
    )
)

> // Option 2:
> // Set properties using capture
> // This method doesn't have the side-effect of associating all future events to the group

Reviewer: ```suggestion
// Option 2: Set properties using capture
// This method doesn't have the side-effect of associating the session's events to the group
```
